### DISCUSSION TITLE
Add As Disassembler Evasion?

### DISCUSSION THREAD

**Discussion by utkonos** [MBCProject/mbc-markdown#157] [Ideas]
This is in a packer that I am analyzing at the moment. There is a lot of code in the packer that is never executed. It's all dead/junk/decoy code not actually part of the packer. It is all made to look like legitimate software. In this case, VLC. In many locations in that dead/junk/decoy code are fake function calls which call an address in the middle of the API hash resolver function. This causes the resulting disassembly to be incorrect unless the function at that location is undefined. Once the function has been undefined the real function is revealed.

Technique: `MBC::Anti-Static Analysis::Disassembler Evasion`
Proposed Method: `Fake Function Call [B0012.007]`

Exemplar Sample: `0a7fd836d36ed8e8e9aa7bc41fdc9242333e8469059dec8886b7d935f3651679`

Address of fake function: `0x100019f0`

Broken disassembly:
<img width="632" alt="broken_disassembly" src="https://github.com/user-attachments/assets/38888023-0ef0-42a8-b768-53b7623e6f1b">

Calls to fake function from dead code:
<img width="312" alt="calls_to_fake_function" src="https://github.com/user-attachments/assets/42104b51-e9d6-4194-81ca-3a4d6c451143">

API hash resolver function after fix with fake function address highlighted and tagged:
<img width="1009" alt="location_of_disassembler_evasion_after_fix" src="https://github.com/user-attachments/assets/4d3631a6-85dd-4a87-96f1-1bfbdd699ee8">

**Comment by dzbeck** [comment]
Thanks @utkonos - looks like a nice addition! Thanks also for the details and screenshots :)

Here's a draft description - any edits would be appreciated:

"A fake function call is used to call an address which confuses a disassembler and results in incorrect disassembly. Manually undefining a fake function call can enable the disassembler to produce the true code."

**Reply by utkonos** [reply]
> Here's a draft description - any edits would be appreciated:
> 
> "A fake function call is used to call an address which confuses a disassembler and results in incorrect disassembly. Manually undefining a fake function call can enable the disassembler to produce the true code."

Looks good.

**Reply by dzbeck** [reply]
great - thanks @utkonos!
